src/openms_gui/source/VISUAL/Plot2DCanvas.cpp

```diff
@@ -2984,7 +2972,7 @@ namespace OpenMS
       }
 
       // update gradient if the min/max intensity changes
-      if (tmp.getIntensity() < current_layer.getFeatureMap()->getMinInt() || tmp.getIntensity() > current_layer.getFeatureMap()->getMaxInt())
+      if (!current_layer.getFeatureMap()->getRange().RangeIntensity::contains(tmp.getIntensity()))
```

Comment:
yes. Its a parent-class scope.
One could also add new methods to the RangeIntensity Class, i.e. `containsIntensity()`. That would avoid the need for the scoping, but its also more code to write in the RangeManager.

Reply:
I would say it would be a bit more in line with the getMinIntensity() etc. methods.
How much code would you need to add?

Reply:
one method per dimension. Four lines each.
I was acutally thinking about using an old-style preprocessor macro, just to avoid writing the same code 4 times (rt, mz, int, mobility). Not sure if thats the right way to go, but it would safe a lot of typing and testing.

Reply:
What do others think? I think will not write those rangemanagers that often and a nice interface would outweight the extra work here.

Reply:
@jpfeuffer  what is your take on that?

Reply:
Might be better for compilation to have less code, which I am valuing very highly these days.
On the other hand this indeed pretty weird and I wouldn't be able to write that kind of code without looking up namespace rules or finding other usages for the first few times.

Reply:
Ah right :). @cbielow can you move some code from the header in the cpp file

Reply:
I can move some code, but what is moveable are mostly one-line functions.
The majority is templates and thus unmovable anyway.
I suspect compile time will not be affected in any way, but I can do it of course.

I also suspect that adding explicit functions will not influence compile time :)
So shall I add them?

Reply:
Ok. Pretty please? 🥺